src/tasks/models/tasksManager.ts

```diff
@@ -201,62 +209,105 @@ export class TasksManager {
+    updateJobParams = {
+      ...updateJobParams,
+      reason: reason ?? undefined,
+      status: finalizeStatus,
+      parameters: { ...job.parameters, callbackParams, cleanupData },
+    };
 
-      const callbackParams: ICallbackExportResponse = {
-        ...callbackSendParams,
-        roi: job.parameters.roi,
-        status: finalizeStatus,
-        errorReason: reason,
-      };
+    return updateJobParams;
+  }
 
-      updateJobParams = { ...updateJobParams, parameters: { ...job.parameters, callbackParams, cleanupData } };
-      this.logger.info({ finalizeStatus, jobId: job.id, msg: `Updating job finalizing status` });
-    } catch (error) {
-      this.logger.error({ jobId: job.id, err: error, reason: `${(error as Error).message}`, msg: `Could not finalize job` });
-      updateJobParams = {
-        ...updateJobParams,
-        reason: JSON.stringify(error as Error),
-        status: OperationStatus.FAILED,
-        parameters: { ...job.parameters, cleanupData },
-      };
-    } finally {
-      await this.jobManagerClient.updateJob(job.id, updateJobParams);
-    }
+  public async finalizeGPKGFailure(job: JobFinalizeResponse, expirationDateUTC: Date, reason: string): Promise<IUpdateJobBody<IJobExportParameters>> {
+    const cleanupData: ICleanupData = this.generateCleanupEntity(job, expirationDateUTC);
+
+    // generate job finally completion with webhook (callback param) data
+    this.logger.info({ jobId: job.id, msg: `Finalize Failure Job` });
+    // create and sending response to callbacks
+    const callbackSendParams = await this.generateCallbackParam(job, expirationDateUTC, reason);
+
+    await this.sendExportCallbacks(job, callbackSendParams);
+
+    const callbackParams: ICallbackExportResponse = {
+      ...callbackSendParams,
+      roi: job.parameters.roi,
+      status: OperationStatus.FAILED,
+      errorReason: reason,
+    };
+
+    this.logger.info({ reason, jobId: job.id, msg: `Updating job finalizing status for failure job` });
+    const updateJobParams: IUpdateJobBody<IJobExportParameters> = {
+      reason: reason,
+      status: OperationStatus.FAILED,
+      percentage: 0,
+      parameters: { ...job.parameters, callbackParams, cleanupData },
+    };
+
+    return updateJobParams;
+  }
+
+  public async createFinalizeTask(job: JobExportResponse, taskType: string, isSuccess = true, reason?: string): Promise<void> {
+    const operationStatus = isSuccess ? OperationStatus.COMPLETED : OperationStatus.FAILED;
+    const taskParameters: ITaskFinalizeParameters = {
+      reason,
+      exporterTaskStatus: operationStatus,
+    };
+    this.logger.info({ jobId: job.id, operationStatus, msg: `Execute finalized job` });
+
+    const createTaskRequest: CreateFinalizeTaskBody = {
+      type: taskType,
+      parameters: taskParameters,
+      status: OperationStatus.PENDING,
+      blockDuplication: true,
+    };
+    await this.jobManagerClient.enqueueTask(job.id, createTaskRequest);
```

Comment:
protect with try catch and print an info log for task already exists

Reply:
done